### PR TITLE
Added Typescript & some types

### PR DESCRIPTION
**Pull request by Christop406**
**NOTE**: These changes have nothing to do with the front-end. I have simply added some object types.
- Added some of the base types outlined in the README (did not add location)
- Added a few helper functions
- Added scripts to package.json
- Added .gitignore

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { HistoricalEvent } from './model/event';
+import { generateId } from './utilities/generate-id';
+import { printEvent } from './utilities/print-event';
+
+main();
+
+// We can do setup here.
+function main(): void {
+    const myEvent: HistoricalEvent = {
```
Comment (Christop406) [inline src/index.ts]: This is just as an example, to show a possible instantiation of a `HistoricalEvent`.

**Comment by MasoomBadi** [comment]
If this is something that we all agree to work on, we must discuss first what should be used and what should be not. I haven't seen any updates from OP, assuming he is busy with studies as we discussed. Platform we plan, Language we use (including back-end) etc.. must be discussed.

**Comment by ExpDev07** [comment]
> 
> 
> If this is something that we all agree to work on, we must discuss first what should be used and what should be not. I haven't seen any updates from OP, assuming he is busy with studies as we discussed. Platform we plan, Language we use (including back-end) etc.. must be discussed.

I agree. What programming languages and eventual frameworks that should be used (for both front end and back end) must be discussed.

**Comment by Christop406** [comment]
People keep saying we need to discuss the stack, but no one is starting that discussion... so I decided to write a little bit of starter code in TypeScript. If the stack ends up moving away from TypeScript/Node to something else, that's fine, but I left the PR here in case people need inspiration, or want to start with the code I wrote.

**Pull request merged**